app/recorder.js

```diff
@@ -1,12 +1,23 @@
-module.exports = app => (name, options) => {
+const Minimatch = require('minimatch').Minimatch;
+
+module.exports = app => (name, options = {}) => {
+  let onlyPattern;
+
   app.locals.recording = true;
   if (!name) throw new Error('Must provide a recording name.');
 
   app.log(['serve', 'record'], name);
 
+  if (options.only) {
+    // if match is a string, assume it is a url pattern
+    onlyPattern = new Minimatch(options.only);
```

Comment:
@AndersDJohnson, this implementation is incorrect. I assumed instantiating `Minimatch` would return a function... it doesn't, and the docs don't make it very straight forward what the instance is good for: https://github.com/isaacs/minimatch#minimatch-class

Reply:
@ryanricard Looks like an instance of `Minimatch` is an object that includes a function called `match`. So we could grab a reference to `onlyPattern.match` and put that in `recordMeta`. I would do something like this:

```js
let only = options.only;

if (typeof only === 'string') {
  const mm = new Minimatch(options.only);
  only = mm.match.bind(mm);
}

app.locals.recordMeta = {
  name,
  {
    ...options,
    only
  }
}
```

That way in the programmatic API, `only` can be provided as a function instead of a string, so that the consumer could use other logic beyond a `minimatch` glob if they wished.